raiden/transfer/routes.py

```diff
@@ -0,0 +1,36 @@
+from raiden.transfer.state import NODE_NETWORK_REACHABLE, RouteState
+from raiden.utils.typing import ChannelID, List, NodeNetworkStateMap
+
+
+def filter_reachable_routes(
+    route_states: List[RouteState], nodeaddresses_to_networkstates: NodeNetworkStateMap
+) -> List[RouteState]:
+    """ This function makes sure we use reachable routes only. """
+
+    return [
+        route
+        for route in route_states
+        if nodeaddresses_to_networkstates.get(route.next_hop_address) == NODE_NETWORK_REACHABLE
+    ]
+
+
+def exclude_routes_from_channels(
```

Comment:
I avoided to use `filter` because this function is about *removing* items that match a certain condition, while functions like `filter_reachable_routes` are about *keeping* the items that match a condition.

We could also think of using `filter` in the name as long as the rest of the function name could indicate the negative (e.g, `filter_ununsed_channels`), but given that in the function above the only criteria for filtering is the channel id, it seems hard to find a suitable negative-condition name.

Reply:
>  this function is about removing items that match a certain condition

It's the opposite, it removing all items that don't fulfill a certain condition: https://docs.python.org/3.5/library/functions.html#filter

Reply:
You're not "Removing" routes, you're just filtering the routes and returning a new list which contain the ones matching a certain condition.

Reply:
@palango and @rakanalh, I agree on both accounts.

But am I the only that seems the inconsistency between the functions in the module? We have:

 - `filter_reachable_routes`  to be a function that takes a list of routes and returns the ones that _are reachable_, i.e, match *positively* the condition and;
 -  `filter_routes_by_channel_ids` being a function that takes a list of routes and returns the ones that are _not_ on the list of channels, i.e, match *negatively*.

I can make the changes if I am the only one bothered by this inconsistency, but thinking as someone that is browsing the code, I'd find very confusing to have a function called `filter_routes_by_channel_ids` and receive in the result none of the channel ids passed as parameters.

Reply:
To move this forward, i don't mind what name you choose but not `exclude_routes_from_channels` because it doesn't make sense IMO

Reply:
Agreed. I am just really struggling to think of a better name for a function that is basically a list comprehension. Part of the hardest problems in Computer Science, it seems.

I will give some minutes of thought into it and rename.